src/VecSim/spaces/L2/L2_SVE_UINT8.h

```diff
@@ -0,0 +1,139 @@
+/*
+ *Copyright Redis Ltd. 2021 - present
+ *Licensed under your choice of the Redis Source Available License 2.0 (RSALv2) or
+ *the Server Side Public License v1 (SSPLv1).
+ */
+
+#include "VecSim/spaces/space_includes.h"
+#include <arm_sve.h>
+
+// Aligned step using svptrue_b8()
+static inline void L2SquareStep(const uint8_t *&pVect1, const uint8_t *&pVect2, size_t &offset,
+                                svfloat32_t &sum) {
+    svbool_t pg = svptrue_b8();
+    // Note: Because all the bits are 1, the extention to 16 and 32 bits does not make a difference
+    // Otherwise, pg should be recalculated for 16 and 32 operations
+
+    svuint8_t v1_ui8 = svld1_u8(pg, pVect1 + offset); // Load uint8 vectors from pVect1
+    svuint8_t v2_ui8 = svld1_u8(pg, pVect2 + offset); // Load uint8 vectors from pVect2
+
+    svint16_t v1_16_l = svreinterpret_s16(svunpklo_u16(v1_ui8));
+    svint16_t v1_16_h = svreinterpret_s16(svunpkhi_u16(v1_ui8));
+    svint16_t v2_16_l = svreinterpret_s16(svunpklo_u16(v2_ui8));
+    svint16_t v2_16_h = svreinterpret_s16(svunpkhi_u16(v2_ui8));
+
+    // Calculate difference and square for low part
+    svint16_t diff_l = svsub_s16_x(pg, v1_16_l, v2_16_l);
+
+    svint32_t diff32_l_l = svunpklo_s32(diff_l);
+    svint32_t diff32_l_h = svunpkhi_s32(diff_l);
```

Comment:
why is this needed? the square diff of `u8` can fit in a `u16`

Reply:
Because values of diffs might be negative, and while modulo arithmetic make it so that the bits value is correct, when we'll multiply the value will no longer be true.